arangodb-foxx-services/src/main/services/prune-database.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2022 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {aql, db} from '@arangodb'
+
+export function pruneBefore(timestamp) {
+    console.log(`[Spline] Prune data before: ${timestamp}`)
+    const t0 = Date.now()
+
+    // STAGE 1: Cleanup event + edges to plans
+    // Option A - https://github.com/AbsaOSS/spline/issues/684#issuecomment-1311499371
+    const refPlanIds = db._query(aql`
+        FOR p IN progress
+            FILTER p.timestamp < ${timestamp}
+            REMOVE p IN progress
+            FOR po IN progressOf
+                FILTER po._from == p._id
+                REMOVE po IN progressOf
+                RETURN DISTINCT po._to
+    `).toArray()
+
+    const t1 = Date.now()
+    console.log(`Purged ${refPlanIds.length} plans, ${t1 - t0} ms`)
```

Comment:
Let's use `Logger` helper object instead of `console` for logging.

```typescript
    import * as Logger from '../utils/logger'
    ...
    Logger.info(`Purged ${refPlanIds.length} plans, ${t1 - t0} ms`)
```

Reply:
Thanks, redone as suggested. I have test-ran it with this logging, too, and witnessed it correctly shows log entries on the ArangoDB LOG UI.